src/core/executor.rs

```diff
@@ -47,16 +49,17 @@ impl Executor {
         match self {
             Executor::SingleThread => args.map(f).collect::<crate::Result<_>>(),
             Executor::ThreadPool(pool) => {
-                let args_with_indices: Vec<(usize, A)> = args.enumerate().collect();
-                let num_fruits = args_with_indices.len();
+                // let args_with_indices: Vec<(usize, A)> = args.enumerate().collect();
+                let args: Vec<A> = args.collect();
+                let num_fruits = args.len();
                 let fruit_receiver = {
                     let (fruit_sender, fruit_receiver) = channel::unbounded();
                     pool.scope(|scope| {
-                        for arg_with_idx in args_with_indices {
+                        for (idx, arg) in args.into_iter().enumerate() {
```

Comment:
what is this AtomicUsize about?

Reply:
The way variables are captured seems to have changed.
I does not compile, because idx is not moved into the closure anymore.
Adding `move` does not work, because it also moves `f`, which we don't want to move.

Reply:
And the AtomicUsize helps!???

Reply:
Yes, with the AtomicUsize, we avoid the lifetime issue. This is the error:

```
error[E0373]: closure may outlive the current function, but it borrows `idx`, which is owned by the current function
  --> src/core/executor.rs:60:41
   |
57 |                     pool.scope(|scope| {
   |                                 ----- has type `&rayon::Scope<'1>`
...
60 |                             scope.spawn(|_| {
   |                                         ^^^ may outlive borrowed value `idx`
61 |                                 let fruit = f(arg);
62 |                                 if let Err(err) = fruit_sender.send((idx, fruit)) {
   |                                                                      --- `idx` is borrowed here
```